libs/cardano-ledger-core/src/Cardano/Ledger/Coin.hs

```diff
@@ -86,9 +88,23 @@ integerToWord64 c
   | c < 0 = Nothing
   | c > fromIntegral (maxBound :: Word64) = Nothing
   | otherwise = Just $ fromIntegral c
+{-# INLINE integerToWord64 #-}
 
 instance ToCBOR (CompactForm Coin) where
   toCBOR (CompactCoin c) = toCBOR c
 
 instance FromCBOR (CompactForm Coin) where
   fromCBOR = CompactCoin <$> fromCBOR
+
+instance Storable Coin where
+  sizeOf _ = sizeOf (undefined :: Word64)
+  {-# INLINE sizeOf #-}
+  alignment _ = alignment (undefined :: Word64)
+  {-# INLINE alignment #-}
+  poke ptr (Coin c) =
+    case integerToWord64 c of
+      Nothing -> error $ "Impossible Coin overflow: " ++ show c
```

Comment:
ah, this is the unfortunate part you mentioned to me earlier. I don't see a way around it, unfortunately.